### PR TITLE
[F2F-873] - Call /abort when user ends their journey

### PR DESCRIPTION
**Pull request by ViveakV**
### What changed

The following 3 actions will now trigger this BE abort flow:

- On click of the “I want to prove my identity another way” - hyperlink at the bottom of the Landing Page
- On click of continue after selecting "I do not have any of these documents" radio button on the DocumentSelection screen
- On click of continue after selecting "Prove your identity another way" radio button on the Date Error Screen

### Issue tracking
https://govukverify.atlassian.net/browse/F2F-873

## Tests

image

image

image

### CODE REVIEW

File: src/app/f2f/controllers/photoIdSelection.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ class PhotoIdSelectionController extends BaseController {
         }
         case APP.PHOTO_ID_OPTIONS.NO_PHOTO_ID: {
           logger.info(
-            "photo-id-selection: user has selected No ID - redirecting to No ID page",
```
Comment (darrendunford) [inline src/app/f2f/controllers/photoIdSelection.js:91]: is there a "No ID" page that needs to be removed?

Comment (ViveakV) [reply src/app/f2f/controllers/photoIdSelection.js:91]: https://github.com/alphagov/di-ipv-cri-f2f-front/blob/main/src/lib/config.js#L35 

It looked to me like it's an external page outside of F2F but not 100%

File: src/app/f2f/controllers/abort.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  }
+  async abortJourney(req) {
+    const headers = {
+      "x-govuk-signin-session-id": req.session.tokenId
+    };
+
+		const resp = await req.axios.post(`${API.PATHS.ABORT}`, {}, {
+      headers,
+    });
```
Comment (darrendunford) [inline src/app/f2f/controllers/abort.js:23]: there is no try/catch around this. When the POST request returns 4XX or 5XX this results in an unhandled exception causing the whole container instance to crash, and ECS has to destroy it and start up a new container instance

File: src/app/f2f/controllers/abort.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  
+
+  it("should be an instance of BaseController", () => {
+    expect(abortController).to.be.an.instanceOf(BaseController);
+  });
+
+	describe("#abortJourney", () => {
+    context("when journey is aborted", () => {
+      it("should call abort endpoint", async () => {
```
Comment (darrendunford) [inline src/app/f2f/controllers/abort.test.js:34]: can we have a negative test - i.e. what happens when the axios call fails

**Review by darrendunford — CHANGES_REQUESTED** [review]

File: src/app/f2f/controllers/abort.js (hunk 2 above)
Comment (ViveakV) [reply src/app/f2f/controllers/abort.js:23]: Will it not be covered by the catch block on line 12 above?

**Review by darrendunford — APPROVED** [review]

**Pull request merged**